src/group_theory/subgroup.lean

```diff
@@ -777,6 +777,9 @@ as additive groups"]
 def prod_equiv (H : subgroup G) (K : subgroup N) : H.prod K ≃* H × K :=
 { map_mul' := λ x y, rfl, .. equiv.set.prod ↑H ↑K }
 
+instance [subsingleton G] : subsingleton (subgroup G) :=
+subsingleton_of_bot_eq_top ((eq_bot_iff_forall ⊤).mpr (λ x _, subsingleton.elim x _)).symm
```

Comment:
Can you add this for `subsingleton (subalgebra R A)` too? Or does that already exist?

Reply:
IN fact, probably you should add `nontrivial` versions of all of these via `not_nontrivial_iff_subsingleton`

Reply:
I opened #5690, which might give some ideas of other bits that are missing.

Reply:
I've gone ahead and put this lemma into #5690, and generalized it to work for `add_subgroup` too

Reply:
Ah, so should I delete this subgroup instance from this PR?